crates/astria-composer/src/searcher/api.rs

```diff
@@ -0,0 +1,59 @@
+use std::{
+    net::SocketAddr,
+    sync::Arc,
+};
+
+use axum::{
+    extract::State,
+    response::IntoResponse,
+    routing::get,
+    Router,
+};
+use serde::Serialize;
+use tracing::info;
+
+use super::State as SearcherState;
+
+pub(super) async fn run(
+    api_url: SocketAddr,
+    state: Arc<SearcherState>,
+) -> Result<(), hyper::Error> {
+    let api_router = Router::new()
+        .route("/healthz", get(healthz))
+        .with_state(state);
+
+    info!(?api_url, "starting api server");
+
+    axum::Server::bind(&api_url)
+        .serve(api_router.into_make_service())
+        .await
+}
+
+pub(super) enum Healthz {
+    Ok,
+    _Degraded,
+}
+
+impl IntoResponse for Healthz {
+    fn into_response(self) -> axum::response::Response {
+        #[derive(Debug, Serialize)]
+        struct HealthzBody {
+            status: &'static str,
+        }
+        let (status, msg) = match self {
+            Self::Ok => (axum::http::StatusCode::OK, "ok"),
+            Self::_Degraded => (axum::http::StatusCode::GATEWAY_TIMEOUT, "degraded"),
+        };
+        let mut response = axum::Json(HealthzBody {
+            status: msg,
+        })
+        .into_response();
+        *response.status_mut() = status;
```

Comment:
I think you might be able to call `(status, json_body).into_response()` because of this: https://docs.rs/axum/latest/axum/response/trait.IntoResponse.html#impl-IntoResponse-for-(StatusCode,+R)